src/main/CommandLine.cpp

```diff
@@ -797,6 +801,53 @@ runRebuildLedgerFromBuckets(CommandLineArgs const& args)
     });
 }
 
+int
+runSimulate(CommandLineArgs const& args)
+{
+    CommandLine::ConfigOption configOption;
+    uint32_t desiredOps = 0;
+    uint32_t startLedger = 0;
+    uint32_t stopLedger = 0;
+    std::string historyArchiveGet;
+    return runWithHelp(
+        args,
+        {configurationParser(configOption),
+         clara::Opt{desiredOps, "DESIRED-OPERATIONS"}["--desired-operations"](
+             "desired number of operations per ledger, will never be less but "
+             "could be up to to 100 more"),
+         clara::Opt{startLedger, "START-LEDGER"}["--start-ledger"](
+             "ledger to start reading from history archive"),
+         clara::Opt{stopLedger, "STOP-LEDGER"}["--stop-ledger"](
+             "ledger to stop reading from history archive")},
+        [&] {
+            auto config = configOption.getConfig();
+            config.setNoListen();
+
+            VirtualClock clock(VirtualClock::REAL_TIME);
+            auto app = Application::create(clock, config, false);
+            app->start();
+
+            LedgerRange lr{startLedger, stopLedger};
```

Comment:
maybe perform some sanity check on the ledger range? This will crash if start = 0 or start > stop

Reply:
Done.

Reply:
I think the new code is not exactly right: you're only calling the validation code when parsing `firstLedgerInclusive`. I think it would be better instead have a `lastLedgerParser` that takes the same validation lambda (I suspect that's what you wanted to do).

Reply:
Validation first parses the entire command, then runs the validator for each parser:
https://github.com/stellar/stellar-core/blob/e93ea152aba73c2ed6d6fb27d15dd9ff6884dedf/src/main/CommandLine.cpp#L221-L237
So there isn't any advantage to having it in both parsers